### ISSUE TITLE
[Crash] Crash when paring instrument to looper and click in record

### ISSUE THREAD

**Issue by lostopkk** [StavWasPlayZ/Even-More-Instruments#7]
Fabric version - 3.0v

[23:02:36] [Render thread/ERROR]: Reported exception thrown!
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:414) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:169) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:102) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:169) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:?]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:?]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403) ~[lwjgl-glfw-3.3.1.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:201) ~[client-intermediary.jar:?]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:219) ~[client-intermediary.jar:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:288) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1241) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[minecraft-1.20.1-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.2.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.2.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.2.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[NewLaunch.jar:?]
Caused by: java.lang.NoSuchMethodError: 'net.minecraft.class_2540 net.minecraft.class_2540.method_52964(boolean)'
	at com.cstav.evenmoreinstruments.networking.packet.LooperRecordStatePacket.write(LooperRecordStatePacket.java:41) ~[evenmoreinstruments-fabric-1.20-1.20.2-3.0.jar:?]
	at net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking.send(ClientPlayNetworking.java:384) ~[fabric-networking-api-v1-1.3.10+eeb8eb3677-5f0732b3ac824797.jar:?]
	at com.cstav.evenmoreinstruments.networking.ModPacketHandler.sendToServer(ModPacketHandler.java:53) ~[evenmoreinstruments-fabric-1.20-1.20.2-3.0.jar:?]
	at com.cstav.evenmoreinstruments.client.gui.instrument.LooperOverlayInjector.onRecordPress(LooperOverlayInjector.java:115) ~[evenmoreinstruments-fabric-1.20-1.20.2-3.0.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:56) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:189) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:409) ~[client-intermediary.jar:?]
	... 19 more

**Comment by sleepy-evelyn** [comment]
Had a similar issue but the looper doesn't load the GUI at all

**Logs**: https://gist.github.com/sleepy-evelyn/c7505058f7da3ef784377c1291398783

**Comment by StavWasPlayZ** [comment]
Thank you both for the report, and sorry for the late reply! Will be looking into it the coming days.

**Comment by StavWasPlayZ** [comment]
@sleepy-evelyn - I see you are using the Quilted Fabric API? The log seems to indicate that a certain method from the Fabric API does not exist. Does this happen in regular Fabric? Do newer versions solve this predicament?  
The specific method in question (should be directed to the folks who manage this library):
```java
java.util.List net.fabricmc.fabric.impl.client.screen.ScreenExtensions.fabric_getButtons()
```

@lostopkk - Not enough context within your report. Please provide the full `debug.log` if still applicable.  
What I'm getting here is that your client does not have an instance of the `PacketByteBuf` class - which is extraordinarily odd, because it is inherently a Minecraft class. I can only assume you are either using the incorrect JAR for your Minecraft installation, or using some sort of a modified client or mod that revises upon this class. Same case scenario here - nothing much I can do about it...

I will be closing this issue for the time being.  
Please do re-open it if you seem to have a lead on the matter, and believe I am at fault.

Have a pleasant day ahead! ☺

**Comment by sleepy-evelyn** [comment]
Possibly. I have yet to come across an issue that's specifically related to Quilt but i'll try the latest version with QuiltMC and Fabric on it's own then test in my modpack.

**Comment by StavWasPlayZ** [comment]
Keep me posted!

**Comment by 1Knightey1** [comment]
Hello there! Stumbled upon your mods when was making a modpack to play with my friends
Everything was fine until my friend decided to test out each and every function the mods have and placed a looper. The world had crashed with no chance to fix (without deleting the mod, that is).
Then I decided to try out myself and also got a crash. Beside that, I have a log now, which I hereby show you:
[crash-2024-07-04_00.24.23-server.txt](https://github.com/user-attachments/files/16091133/crash-2024-07-04_00.24.23-server.txt)
I hope there is a chance to fix this, because my friend really liked this mod and I don't want to exclude it from the modpack.

**Comment by StavWasPlayZ** [comment]
Hey there! Thank you for your report.  
This doesn't seem to be an issue relating to the current issue thread; so I will dedicate it as an independent issue of its own. I'd appreciate if you could hit its 'follow' button when you are able.

I see the version you are playing is 1.20.4. Could you perhaps try to reproduce the issue in Minecraft 1.20.1? It might be possible I did not test enough for this specific version.

Nevertheless, I shall further delve into this issue when I'm more free, perhaps next week or so.  
Waiting for your update, and apologies for the inconvenience!

**Comment by StavWasPlayZ** [comment]
I will further lock the current thread, and will assume it solved per v6.0 for lack of activity.
